backend/de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Conditions_Extend.java

```diff
@@ -0,0 +1,107 @@
+package de.metas.contracts.flatrate.process;
+
+import de.metas.calendar.standard.YearAndCalendarId;
+import de.metas.calendar.standard.YearId;
+import de.metas.contracts.ConditionsId;
+import de.metas.contracts.IFlatrateBL;
+import de.metas.contracts.IFlatrateDAO;
+import de.metas.contracts.model.I_C_Flatrate_Conditions;
+import de.metas.contracts.model.I_ModCntr_Settings;
+import de.metas.contracts.modular.settings.ModularContractSettingsDAO;
+import de.metas.contracts.modular.settings.ModularContractSettingsQuery;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.IProcessPreconditionsContext;
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+import de.metas.process.ProcessExecutionResult.RecordsToOpen.OpenTarget;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.product.ProductId;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_C_Year;
+import org.compiere.util.Ini;
+
+import static de.metas.contracts.model.X_C_Flatrate_Conditions.TYPE_CONDITIONS_ModularContract;
+import static de.metas.contracts.model.X_C_Flatrate_Conditions.ONFLATRATETERMEXTEND_ExtensionNotAllowed;
+
+public class C_Flatrate_Conditions_Extend extends JavaProcess implements IProcessPrecondition
```

Comment:
Please change terminology here, since we're not extending existing contract conditions, but in fact cloning them.

Reply:
also we should make it clear that this process is only relevant for modular contracts.
How about `C_Flatrate_Conditions_ModularContract_Clone`?

Reply:
agree